app/controllers/student_requests_controller.rb

```diff
@@ -12,6 +12,9 @@ def student_request_params
   end
 
   def index
+    if session_get(:uin) == nil
+      redirect_to root_path
+    end
```

Comment:
Better to have if else when using redirect_to. The control also continues beyond redirect_to and may cause inconsistency.